### DISCUSSION TITLE
Saving forecast object

### DISCUSSION THREAD

**Discussion by philipmassie** [Nixtla/mlforecast#116] [Q&A] [answered]
Hello
Really enjoying using this library, thank you for all the work!
I'm probably missing it somewhere, but is there an approach for saving the forecast object along with fitted model(s), feature functions and so on? There's so much going on in there, it seems like it might not be reasonable.
Thanks in advance
Phil

Update: I see how to save the model objects out of the forecast object, but not how to add the pretrained models back to the object.

**Comment by jmoralez** [comment, marked as answer]
Hi. We're glad you're finding the library useful.

We don't have any built-in functionalities to save and load, however you should be able to use pickle/cloudpickle to do so, e.g.
```python
import cloudpickle
from mlforecast import MLForecast

mlf = MLForecast(...)
mlf.fit(...)
with open('mlf.pkl', 'wb') as f:
    cloudpickle.dump(mlf, f)

# on a separate session
with open('mlf.pkl', 'rb') as f:
    mlf = cloudpickle.load(f)
```

Does your use case require some other type of serialization?

**Reply by philipmassie** [reply]
Thanks for the response. 
No I dont think we'd need anything special. 
I suspect pickle didn't work because of my custom date functions, but I'll investigate further and report back if I find out.

**Reply by jmoralez** [reply]
cloudpickle should be able to serialize those but feel free to open an issue if you run into any problems so we can take a look.

**Reply by philipmassie** [reply]
Thank you so much, cloudpickle did the trick.
